### ISSUE TITLE
Adding new type should make the form dirty

### ISSUE THREAD

**Issue by wbazant** [falling-fruit/falling-fruit-web#553]
Types starts off red. Types remains red after adding a new type.

Loading form:

<img src="https://github.com/user-attachments/assets/4afd0e73-d314-4615-8cee-b40d05ab4cc3" width="300px">

Adding a new type:

<img src="https://github.com/user-attachments/assets/28719e73-a08d-4456-b690-826cb5f4dfb2" width="300px">

Adding an existing type:

<img src="https://github.com/user-attachments/assets/36c3adb8-2da1-4c7d-ab13-c21b892a1954" width="300px">

_Originally posted by @ezwelty in https://github.com/falling-fruit/falling-fruit-web/issues/547#issuecomment-2449421041_

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've identified the issue causing types to remain red after adding a new type. The problem was in the `handleNewType` function in the `TypesSelect` component. The function was not properly updating the form values. I've updated the `handleNewType` function to correctly set the new type in the form values.

Here's a summary of the changes I made:

1. In the `TypesSelect` component, I modified the `handleNewType` function to correctly update the form values.
2. I also added a check to ensure that the new type is not already present in the selected types.

These changes should resolve the issue of types remaining red after adding a new type.

I also created pull request: https://github.com/falling-fruit/falling-fruit-web/pull/554


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
